### ISSUE TITLE
Allow `cy.visit` to work on 404 pages (option `failOnStatusCode`)

### ISSUE THREAD

**Issue by Janiczek** [cypress-io/cypress#421]
Right now I can't do any of the `cy.visit()`-enabled BDD on our HTTP 404 pages. I can test it crudely with:
```
cy.request({url: '/abcde', failOnStatusCode: false})
  .its('body').should('.include', 'Oops, something went wrong!');
```
But I _(think I)_ can't do any clicking etc.

If `cy.visit()` options allowed for the `failOnStatusCode: bool` option, I could do something like:

```
cy.visit('/abcde', {failOnStatusCode: false});
cy.get('#email').type('my@email.com');
cy.get('#submit').click();
cy.url().should('contain','/form-success');
```

**Comment by brian-mann** [comment]
Yes, we can add this for sure. 

404 is a perfectly legit status code and often webservers send some special page which you should be able to test for.

**Comment by mellogarrett** [comment]
Just curious on the status of this. I'd love to be able to set the expected status code manually. When developing & using `python -m SimpleHTTPServer`, it returns code `304 Not Modified` which `.visit` fails on right now.

**Comment by brian-mann** [comment]
Are you sure it fails on `304`? In our code we use this regexp:

`isOkStatusCodeRe = /^[2|3]\d+$/`

`3xx` should not fail the visit.

**Comment by mellogarrett** [comment]
my test looks like this: 
```
describe("tests", () => {
  context("Actions", () => {
    beforeEach(() => {
      cy.visit("localhost:5000");
    });

    it("Should check Title", () => {
      cy.title().should("equal", "React App");
    });
  });
});
```

<img width="1440" alt="screen shot 2017-10-09 at 11 31 58 am" src="https://user-images.githubusercontent.com/9908831/31345997-930fb842-ace5-11e7-8b35-8cd3b1d06279.png">

**Comment by bahmutov** [comment]
Does it fail if you use explicit protocol in the URL like "http://localhost:5000"?

**Comment by mellogarrett** [comment]
@bahmutov Yea, same result either way.

**Comment by mellogarrett** [comment]
Hmm, this kinda prevents me from doing anything in Cypress. Should I create a new issue? Any ideas?

**Comment by brian-mann** [comment]
@mellogarrett this is not coming from Cypress. This is an error **literally** coming from your application, which is why Cypress is failing. In other words, this is Cypress catching bugs in your own code.

By default Cypress automatically catches global uncaught exceptions.

Read this issue for more detail: https://github.com/cypress-io/cypress/issues/254

You can also turn off uncaught exception handling here: https://docs.cypress.io/api/events/catalog-of-events.html

**Comment by mellogarrett** [comment]
@brian-mann you were totally right (of course, you already knew that...) 

Thanks a ton for working through this with me. I'm super excited to start using Cypress!

**Comment by bahmutov** [comment]
Brian, good call

**Comment by brian-mann** [comment]
Fixed by https://github.com/cypress-io/cypress/pull/982

**Comment by brian-mann** [comment]
Released in [`1.1.3`](https://docs.cypress.io/guides/references/changelog.html#1-1-3).

**Comment by toastcrunch** [comment]
beforeEach(() => {
    cy.visit('https://2u.netlify.com/', {failOnStatusCode: false}) //Bypass 4xx authorization
    cy.get('input').type('xxx').type('{enter}')
}
